source/gui/settingsDialogs.py

```diff
@@ -84,6 +82,10 @@ def __new__(cls, *args, **kwargs):
 		SettingsDialog._instances.add(obj)
 		return obj
 
+	def _removeOpenDlg(self):
+		if self in SettingsDialog._instances:

```

Comment:
I'm not sure whether I already expressed my concerns with this approach. I'm afraid that this way of forcefully destroying instances will obfuscate cases where a dialog isn't destroyed properly due to bugs, such as #8372.

Reply:
I do share this concern, there are more complicated things we could do perhaps but I don't want to go down that path right now. This approach will help to stop issues that require the user to restart before they can re-open the settings dialog. Whereas removing it, might make it easier to catch circular references. Essentially there are two purposes. Rather than blocking users from opening the settings dialog, I would prefer a log error in the case of a circular reference detection.

Reply:
I think I have addressed this concern now. I have updated the PR description to discuss this.